### ISSUE TITLE
Suggest separate gtol ctol trtol xtol in trust-constr

### ISSUE THREAD

**Issue by denis-bz** [scipy/scipy#9628]
A low-priority enhancement request
`gtol` in trust-constr means *both* `|gradient| < gtol` and `constraints < gtol`.
And who would guess that the trust_radius tolerance is called `xtol` ?
While these are documented on the web and in the source
and easy enough to do in a callback,
I'd find separate `gtol ctol trtol xtol` clearer.

(Fwiw, `gtol` bit me in 1.2:
in 1.1, "gtol termination condition is satisfied" (actually ctol)
but 1.2 kept going to maxiter, got stuck, on a non-convex test problem ABXC:
I guess tiny differences between mac Accelerate and openblas, don't know.)

**Comment by donhatch** [comment]
I'm a bit confused...
You're suggesting having *both* trtol and xtol?  What would their meanings be?

**Comment by denis-bz** [comment]
@DonHatch, you're right, what I should've said is that `trtol` would be a better name than `xtol`(not both). \
Anyway, after 8 years, close.
